pims/cine.py

```diff
@@ -288,13 +358,52 @@ def __init__(self, filename):
                                                    })
         self.stack_meta_data['trigger_time'] = self.trigger_time
 
+        ### IMAGES
+        # TODO: add support for reading sequence within the same framework, when data
+        # has been saved in another format (.tif, image sequence, etc)
+        if splitext(filename)[-1] in ['.cine', '.cci', '.cin']:
```

Comment:
A different test based on file size or contents would be much better. File extensions can be unreliable, and in general PIMS only uses extensions for the automatic `open()` feature, not within individual readers. At minimum, this test should be case-insensitive.

Reply:
Such as testing the content after the image offset? I'll propose some stuff based on that idea.

Reply:
Great! Yes, I think comparing the total file size to the image offset would work well. I am thinking of use cases where people are reading from object stores or other file-like objects where the extension is not always given.

It just occurs to me: isn't a .chd file just a .cine file with zero images? That should not cause any problem for the code and would be easy to detect with the existing `__len__()` implementation.